crates/revm/src/evm_impl.rs

```diff
@@ -125,6 +125,8 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
 
     fn transact_preverified(&mut self) -> EVMResult<DB::Error> {
         let env = &self.data.env;
+        #[cfg(feature = "memory_limit")]
```

Comment:
I don't think we need this feature anymore if we're going to have a limit always? cc @rakita

Reply:
This enables the env field in the first place, but if we're going to store a limit in the memory I don't see a reason to cfg this

Reply:
My first idea was to keep the feature memory limit as a cap of how much memory you'd like to allocate, since the upper bound calculation can return a potentially high number of bytes. An example was here:
```rust
pub fn new_with_memory_limit(gas_limit: u64, memory_limit: u64) -> Self {
        let size = min(
            SharedMemory::calculate_upper_bound(gas_limit) as usize,
            memory_limit as usize,
        );
```

 If you want I can remove it, no problem!